src/System.IO.Pipelines.Extensions/UnownedBufferReader.cs

```diff
@@ -66,6 +62,11 @@ public UnownedBufferReader()
 
         bool IReadableBufferAwaiter.IsCompleted => IsCompleted;
 
+        bool IReadableBufferAwaiter.IsCompletedSuccessfully => 
+            IsCompleted && 
+            _cancelledState == CancelledState.NotCancelled && 
+            !Reading.IsCanceled && !Reading.IsFaulted;
```

Comment:
Trouble is this will throw rather than returning a faulted task https://github.com/dotnet/corefxlab/blob/master/src/System.IO.Pipelines.Extensions/PipelineReaderExtensions.cs#L11-L22